pkg/etcdcli/etcdcli.go

```diff
@@ -49,27 +51,24 @@ type etcdClientGetter struct {
 
 func NewEtcdClient(kubeInformers v1helpers.KubeInformersForNamespaces, networkInformer configv1informers.NetworkInformer, eventRecorder events.Recorder) EtcdClient {
 	return &etcdClientGetter{
-		nodeLister:            kubeInformers.InformersFor("").Core().V1().Nodes().Lister(),
-		endpointsLister:       kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Lister(),
-		networkLister:         networkInformer.Lister(),
-		nodeListerSynced:      kubeInformers.InformersFor("").Core().V1().Nodes().Informer().HasSynced,
-		endpointsListerSynced: kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Informer().HasSynced,
-		networkListerSynced:   networkInformer.Informer().HasSynced,
-		eventRecorder:         eventRecorder.WithComponentSuffix("etcd-client"),
+		nodeLister:             kubeInformers.InformersFor("").Core().V1().Nodes().Lister(),
+		configmapsLister:       kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Lister(),
+		networkLister:          networkInformer.Lister(),
+		nodeListerSynced:       kubeInformers.InformersFor("").Core().V1().Nodes().Informer().HasSynced,
+		endpointsListerSynced:  kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Informer().HasSynced,
+		configmapsListerSynced: kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer().HasSynced,
+		networkListerSynced:    networkInformer.Informer().HasSynced,
+		eventRecorder:          eventRecorder.WithComponentSuffix("etcd-client"),
 	}
 }
 
 // getEtcdClient may return a cached client.  When a new client is needed, the previous client is closed.
 // The caller should not closer the client or future calls may fail.
 func (g *etcdClientGetter) getEtcdClient() (*clientv3.Client, error) {
-	if !g.nodeListerSynced() {
-		return nil, fmt.Errorf("node lister not synced")
-	}
-	if !g.endpointsListerSynced() {
-		return nil, fmt.Errorf("node lister not synced")
-	}
-	if !g.networkListerSynced() {
-		return nil, fmt.Errorf("network lister not synced")
+	// TODO: inject done channel from context
+	if !cache.WaitForCacheSync(make(<-chan struct{}),
```

Comment:
the previous structure was intentional to avoid blocking the control loops.  The flat out failure is very easy to reason about based on the messages.  A wait is less easy to reason about.

Reply:
Reverted